src/current/_includes/v23.1/start-in-docker/mac-linux-steps-single-node.md

```diff
@@ -59,12 +59,10 @@ The `cockroach` process listens on `127.0.0.1:26257` and `localhost:26257`, and
       --env COCKROACH_PASSWORD={PASSWORD} \
       --name=roach-single \
       --hostname=roach-single \
-      --net=roachnet \
       -p 26257:26257 \
       -p 8080:8080 \
       -v "roach-single:/cockroach/cockroach-data"  \
-      {{page.release_info.docker_image}}:{{page.release_info.version}} start-single-node \
-      --http-addr=localhost:8080 \
+      {{page.release_info.docker_image}}:{{page.release_info.version}} start-single-node
```

Comment:
I prefer for you to revert this line, because we should specify where DB Console should listen. The `--listen-addr` flag is ignored for single-node deployments, but `--http-addr` is not, if I understand correctly.

Reply:
When 
```
$ podman run -it --rm --network container:roach-single ubuntu:latest bash -c "apt update && apt install -y iproute2 && ss -plnt"
...
State          Recv-Q         Send-Q                  Local Address:Port                    Peer Address:Port         Process
LISTEN         0              0                           127.0.0.1:8080                         0.0.0.0:*
LISTEN         0              0                                   *:26257                              *:*
```

```

State          Recv-Q         Send-Q                  Local Address:Port                    Peer Address:Port         Process
LISTEN         0              0                                   *:8080                               *:*
LISTEN         0              0                                   *:26257                              *:*
```

Reply:
Looks like `http-address` is also ignored in single mode. @ZhouXing19 Can you weigh in here? I believe you did some work on this previously: https://github.com/cockroachdb/cockroach/pull/70238

Reply:
Jane is going to look at this tomorrow, but from looking at https://github.com/ZhouXing19/cockroach/blob/master/build/deploy/cockroach.sh it looks like you can specify a different advertise address but if you specify a different advertise port, we throw it away and stubbornly use 8080.

Reply:
Thank you for the spelunking! I'll wait for Jane to weigh in so we can discuss this further.

Reply:
Thanks for reporting this issue. In the current implementation `http-address` should be resolved and applied. 
But I'm not 100% sure if I understand the problem here correctly, and my following test seems to confirm my idea:

```
❯ docker run -d --name=roach5 --hostname=roach5 -p 26258:26257 -p 8081:8080   cockroachdb/cockroach-local:latest start-single-node --insecure --http-addr=localhost:8081
0c19b937246131ce1bcc26551f59cfd24f5b212a238336b9fec45a85cabd8dfa

❯ nc -vz localhost 26257         
nc: connectx to localhost port 26257 (tcp) failed: Connection refused
nc: connectx to localhost port 26257 (tcp) failed: Connection refused


❯ nc -vz localhost 26258
nc: connectx to localhost port 26258 (tcp) failed: Connection refused
Connection to localhost port 26258 [tcp/*] succeeded!


❯ nc -vz localhost 8080 
nc: connectx to localhost port 8080 (tcp) failed: Connection refused
nc: connectx to localhost port 8080 (tcp) failed: Connection refused

❯ nc -vz localhost 8081
nc: connectx to localhost port 8081 (tcp) failed: Connection refused
Connection to localhost port 8081 [tcp/sunproxyadmin] succeeded!
```

So it seems that we did exposed the http port via 8081, and it is accessible.

Reply:
Thank you for getting back to us. I believe the `nc` test is hiding the issue since `nc`'s `-z` option will only report the connection status without establishing a connection. If you remove the `-z` option you'll see that the TCP connection gets reset and control is returned immediately back to your terminal. This connection reset can be seen using `curl` as well.

```sh
$ docker run -d --name=roach-single --hostname=roach-single -p 3000:26257 -p 4000:8080 docker.io/cockroachdb/cockroach:v23.1.11 start-single-node --insecure --http-addr=localhost:8080
f0ab87b54fd5fb18bb5275bc8c97aa2db099ef11add7e3e1bb7570d5771f943c

$ nc -v localhost 4000
Connection to localhost port 4000 [tcp/terabase] succeeded!

$ curl -vL http://localhost:4000
*   Trying 127.0.0.1:4000...
* Connected to localhost (127.0.0.1) port 4000 (#0)
> GET / HTTP/1.1
> Host: localhost:4000
> User-Agent: curl/8.1.2
> Accept: */*
>
* Recv failure: Connection reset by peer
* Closing connection 0
curl: (56) Recv failure: Connection reset by peer
```

I believe this is because the host's `4000` binding cannot communicate with the container's `8080` binding since the container's binding is listening on `localhost`, not `*`.

If we remove the `--http-addr` option, the container's `8080` binding will listen on `*` instead of `localhost`, and the host's `4000` binding can communicate with the container's binding. With this change, I can now use `nc` to make a raw HTTP request and then quit with Control-D. Using `curl` also now works.

```sh
$ docker run -d --name=roach-single --hostname=roach-single -p 3000:26257 -p 4000:8080 docker.io/cockroachdb/cockroach:v23.1.11 start-single-node --insecure --http-addr=localhost:8080
2bf0084bcd511cd2df59cccecd97e85c60423da93d3b0a0bb38ecee24763e74b

> nc -v localhost 4000
Connection to localhost port 4000 [tcp/terabase] succeeded!
GET / HTTP/1.1
Host: localhost:4000

HTTP/1.1 200 OK
Accept-Ranges: bytes
Cache-Control: no-cache
Content-Length: 264
Content-Type: text/html; charset=utf-8
Last-Modified: Wed, 27 Sep 2023 02:36:23 GMT
Vary: Accept-Encoding
Date: Thu, 09 Nov 2023 22:58:39 GMT

<!DOCTYPE html>
<html>
        <head>
                <title>Cockroach Console</title>
                <meta charset="UTF-8">
                <link href="favicon.ico" rel="shortcut icon">
        </head>
        <body>
                <div id="react-layout"></div>
                <script src="bundle.js" type="text/javascript"></script>
        </body>
</html>
^D

$ curl -vL http://localhost:4000
*   Trying 127.0.0.1:4000...
* Connected to localhost (127.0.0.1) port 4000 (#0)
> GET / HTTP/1.1
> Host: localhost:4000
> User-Agent: curl/8.1.2
> Accept: */*
>
< HTTP/1.1 200 OK
< Accept-Ranges: bytes
< Cache-Control: no-cache
< Content-Length: 264
< Content-Type: text/html; charset=utf-8
< Last-Modified: Wed, 27 Sep 2023 02:36:23 GMT
< Vary: Accept-Encoding
< Date: Thu, 09 Nov 2023 22:55:20 GMT
<
<!DOCTYPE html>
<html>
        <head>
                <title>Cockroach Console</title>
                <meta charset="UTF-8">
                <link href="favicon.ico" rel="shortcut icon">
        </head>
        <body>
                <div id="react-layout"></div>
                <script src="bundle.js" type="text/javascript"></script>
        </body>
</html>
* Connection #0 to host localhost left intact
```

Reply:
Thanks for detailed explanation. Yes, it makes sense to me to not use `--http-addr=localhost:8080`. However, `--http-addr=roach-single:8080` or `--http-addr=0.0.0.0:8080` should work. 

```
❯ docker run -d --name=roach-single --hostname=roach-single -p 3000:26257 -p 4000:8081 cockroachdb/cockroach:latest-v23.1 start-single-node --insecure --http-addr=roach-single:8081 
1e9ead570cf6978e66d7d3addc00602338f38105fcfd0e4d38e0096be13bfb57

❯ curl -vL http://127.0.0.1:4000
*   Trying 127.0.0.1:4000...
* Connected to 127.0.0.1 (127.0.0.1) port 4000 (#0)
> GET / HTTP/1.1
> Host: 127.0.0.1:4000
> User-Agent: curl/7.84.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Accept-Ranges: bytes
< Cache-Control: no-cache
< Content-Length: 264
< Content-Type: text/html; charset=utf-8
< Last-Modified: Wed, 27 Sep 2023 01:53:43 GMT
< Vary: Accept-Encoding
< Date: Mon, 13 Nov 2023 18:52:07 GMT
< 
<!DOCTYPE html>
<html>
        <head>
                <title>Cockroach Console</title>
                <meta charset="UTF-8">
                <link href="favicon.ico" rel="shortcut icon">
        </head>
        <body>
                <div id="react-layout"></div>
                <script src="bundle.js" type="text/javascript"></script>
        </body>
</html>
* Connection #0 to host 127.0.0.1 left intact
```

Essentially, IIUC the key here is that though`--http-addr` flag is _not_ ignored for the `start-single-node` command, the port forwarding supported by docker always maps a TCP port inside a container, rather than an ip-specific port. The docker doc I found didn't explicitly point this out, but it seems to be [a commonly recommended practice](https://stackoverflow.com/a/57427805/22315968) to let the app inside the docker bind itself to 0.0.0.0 rather than 127.0.0.1.

Reply:
Thank you for confirming! Given this information, I think it's best to use `--http-addr=roach-single:8080` in the documentation.

@mdlinville what do you think?